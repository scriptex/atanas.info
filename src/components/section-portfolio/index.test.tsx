import * as React from 'react';
import { act } from 'react-dom/test-utils';
import { render } from '@testing-library/react';

import { SectionPortfolio } from '.';

jest.mock('../../data/projects-list.json', () => ({
	default: [
		{
			url: 'https://app.home.eon.com/',
			image: 'images/temp/eon-home.png',
			title: 'E.ON Home',
			description: 'TypeScript, SCSS, React, Redux, MS Azure, NodeJS'
		},
		{
			url: 'https://hems-admin-portal.eon.ooo/',
			image: 'images/temp/eon-hems.png',
			title: 'E.ON H.E.M.S (Home Energy Management System)',
			description: 'TypeScript, SCSS, Angular, Redux, MS Azure, NodeJS',
			timeout: 5000
		},
		{
			url: 'https://xpndnow.com/',
			image: 'images/temp/xpnd.png',
			title: 'XPND',
			description: 'TypeScript, SCSS, React, Redux, NodeJS'
		},
		{
			url: 'https://xpndnow.com/ebook',
			image: 'images/temp/xpnd-ebook.png',
			title: 'XPND Ebook',
			description: 'TypeScript, SCSS, React, Redux, NodeJS'
		},
		{
			url: 'https://demo.xpndnow.com/',
			image: 'images/temp/xpnd-demo.png',
			title: 'XPND demo application',
			description: 'TypeScript, SCSS, React, Redux, NodeJS'
		},
		{
			url: 'http://emailio.com/',
			image: 'images/temp/emailio.png',
			title: 'Emailio Web Application',
			description: 'SCSS, JavaScript, React, Redux, Redux Saga'
		},
		{
			url: 'https://www.nulla.tv/',
			image: 'images/temp/nulla-tv.png',
			title: 'Nulla TV',
			description: 'HTML5, CSS3, JavaScript, WordPress'
		},
		{
			url: 'https://kinetikautomotive.com/',
			image: 'images/temp/kinetik-automotive.png',
			title: 'Kinetik Automotive',
			description: 'HTML5, CSS3, JavaScript, Vue, Nuxt, WordPress'
		},
		{
			url: 'https://www.sod.bg/',
			image: 'images/temp/sod-bg.png',
			title: 'SOD 64 (СОД 64)',
			description: 'HTML5, CSS3, JavaScript, jQuery, PHP'
		},
		{
			url: '',
			image: 'images/temp/at-the-wall.png',
			title: 'At the Wall',
			description: 'CSS3, PHP'
		},
		{
			url: 'https://snake.atanas.info',
			image: 'images/temp/material-snake.png',
			title: 'Material Snake',
			description: 'TypeScript, HTML, CSS3'
		},
		{
			url: 'https://www.google.com',
			image: 'image 1',
			title: 'Project 1',
			description: 'Description 1'
		}
	]
}));

describe('SectionPortfolio component', () => {
	it('Should render the SectionPortfolio component', () => {
		const { asFragment, container } = render(<SectionPortfolio />);

		expect(asFragment()).toMatchSnapshot();

		act(() => {
			container.querySelectorAll<HTMLElement>('.c-section__actions .c-btn').forEach(button => {
				button.click();

				expect(asFragment()).toMatchSnapshot();
			});
		});
	});
});
