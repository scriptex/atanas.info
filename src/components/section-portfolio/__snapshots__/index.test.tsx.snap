// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionPortfolio component Should render the SectionPortfolio component 1`] = `
<Component
  actions={
    <Unknown
      onClick={[Function]}
    >
      Show all
    </Unknown>
  }
  className=" fullsize-background"
  hasButton={true}
  hasShell={true}
  id="portfolio"
  style={
    Object {
      "backgroundImage": "url(images/temp/desktop.jpg)",
    }
  }
>
  <h1>
    Portfolio
  </h1>
  <h2>
    Mobile applications
  </h2>
  <div
    className="c-section__sliders"
  >
    <div
      className="c-section__slider"
      key="0"
    >
      <Component
        href="https://apps.apple.com/gb/app/e-on-home-solar-and-heating/id1468486784#?platform=iphone"
      >
        <h3>
          E.ON Home for iPhone
        </h3>
        <p>
          An award winning smart energy management application for iOS
        </p>
        <code>
          Technologies used:
          <br />
          TypeScript, React, React Native, Redux, MS Azure, AWS, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        responsive={Array []}
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="E.ON Home for iPhone screenshot 1"
            src="images/temp/eon-home-iphone-dark1.png"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="E.ON Home for iPhone screenshot 2"
            src="images/temp/eon-home-iphone-dark2.png"
          />
        </div>
        <div
          key="2"
        >
          <img
            alt="E.ON Home for iPhone screenshot 3"
            src="images/temp/eon-home-iphone-dark3.png"
          />
        </div>
        <div
          key="3"
        >
          <img
            alt="E.ON Home for iPhone screenshot 4"
            src="images/temp/eon-home-iphone-dark4.png"
          />
        </div>
        <div
          key="4"
        >
          <img
            alt="E.ON Home for iPhone screenshot 5"
            src="images/temp/eon-home-iphone-light1.png"
          />
        </div>
        <div
          key="5"
        >
          <img
            alt="E.ON Home for iPhone screenshot 6"
            src="images/temp/eon-home-iphone-light2.png"
          />
        </div>
        <div
          key="6"
        >
          <img
            alt="E.ON Home for iPhone screenshot 7"
            src="images/temp/eon-home-iphone-light3.png"
          />
        </div>
        <div
          key="7"
        >
          <img
            alt="E.ON Home for iPhone screenshot 8"
            src="images/temp/eon-home-iphone-light4.png"
          />
        </div>
      </Slider>
    </div>
    <div
      className="c-section__slider"
      key="1"
    >
      <Component
        href="https://apps.apple.com/us/app/foton-home/id1525688620#?platform=iphone"
      >
        <h3>
          Foton Home for iPhone
        </h3>
        <p>
          Smart energy management application for iOS
        </p>
        <code>
          Technologies used:
          <br />
          TypeScript, React, React Native, Redux, MS Azure, AWS, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        responsive={Array []}
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="Foton Home for iPhone screenshot 1"
            src="images/temp/foton-home-iphone-light1.png"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="Foton Home for iPhone screenshot 2"
            src="images/temp/foton-home-iphone-light2.png"
          />
        </div>
        <div
          key="2"
        >
          <img
            alt="Foton Home for iPhone screenshot 3"
            src="images/temp/foton-home-iphone-light3.png"
          />
        </div>
        <div
          key="3"
        >
          <img
            alt="Foton Home for iPhone screenshot 4"
            src="images/temp/foton-home-iphone-light4.png"
          />
        </div>
        <div
          key="4"
        >
          <img
            alt="Foton Home for iPhone screenshot 5"
            src="images/temp/foton-home-iphone-dark1.png"
          />
        </div>
        <div
          key="5"
        >
          <img
            alt="Foton Home for iPhone screenshot 6"
            src="images/temp/foton-home-iphone-dark2.png"
          />
        </div>
        <div
          key="6"
        >
          <img
            alt="Foton Home for iPhone screenshot 7"
            src="images/temp/foton-home-iphone-dark3.png"
          />
        </div>
        <div
          key="7"
        >
          <img
            alt="Foton Home for iPhone screenshot 8"
            src="images/temp/foton-home-iphone-dark4.png"
          />
        </div>
      </Slider>
    </div>
    <div
      className="c-section__slider"
      key="2"
    >
      <Component
        href="https://play.google.com/store/apps/details?id=com.eon.home.eu"
      >
        <h3>
          E.ON Home for Android
        </h3>
        <p>
          An award winning smart energy management application for Android
        </p>
        <code>
          Technologies used:
          <br />
          TypeScript, React, React Native, Redux, MS Azure, AWS, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        responsive={Array []}
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="E.ON Home for Android screenshot 1"
            src="images/temp/eon-home-android1.webp"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="E.ON Home for Android screenshot 2"
            src="images/temp/eon-home-android2.webp"
          />
        </div>
        <div
          key="2"
        >
          <img
            alt="E.ON Home for Android screenshot 3"
            src="images/temp/eon-home-android3.webp"
          />
        </div>
        <div
          key="3"
        >
          <img
            alt="E.ON Home for Android screenshot 4"
            src="images/temp/eon-home-android4.webp"
          />
        </div>
        <div
          key="4"
        >
          <img
            alt="E.ON Home for Android screenshot 5"
            src="images/temp/eon-home-android5.webp"
          />
        </div>
        <div
          key="5"
        >
          <img
            alt="E.ON Home for Android screenshot 6"
            src="images/temp/eon-home-android6.webp"
          />
        </div>
        <div
          key="6"
        >
          <img
            alt="E.ON Home for Android screenshot 7"
            src="images/temp/eon-home-android7.webp"
          />
        </div>
        <div
          key="7"
        >
          <img
            alt="E.ON Home for Android screenshot 8"
            src="images/temp/eon-home-android8.webp"
          />
        </div>
      </Slider>
    </div>
    <div
      className="c-section__slider"
      key="3"
    >
      <Component
        href="https://play.google.com/store/apps/details?id=com.foton.home.eu"
      >
        <h3>
          Foton Home for Android
        </h3>
        <p>
          Smart energy management application for Android
        </p>
        <code>
          Technologies used:
          <br />
          TypeScript, React, React Native, Redux, MS Azure, AWS, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        responsive={Array []}
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="Foton Home for Android screenshot 1"
            src="images/temp/foton-home-android5.webp"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="Foton Home for Android screenshot 2"
            src="images/temp/foton-home-android6.webp"
          />
        </div>
        <div
          key="2"
        >
          <img
            alt="Foton Home for Android screenshot 3"
            src="images/temp/foton-home-android7.webp"
          />
        </div>
        <div
          key="3"
        >
          <img
            alt="Foton Home for Android screenshot 4"
            src="images/temp/foton-home-android8.webp"
          />
        </div>
        <div
          key="4"
        >
          <img
            alt="Foton Home for Android screenshot 5"
            src="images/temp/foton-home-android1.webp"
          />
        </div>
        <div
          key="5"
        >
          <img
            alt="Foton Home for Android screenshot 6"
            src="images/temp/foton-home-android2.webp"
          />
        </div>
        <div
          key="6"
        >
          <img
            alt="Foton Home for Android screenshot 7"
            src="images/temp/foton-home-android3.webp"
          />
        </div>
        <div
          key="7"
        >
          <img
            alt="Foton Home for Android screenshot 8"
            src="images/temp/foton-home-android4.webp"
          />
        </div>
      </Slider>
    </div>
    <div
      className="c-section__slider"
      key="4"
    >
      <Component
        href="https://apps.apple.com/gb/app/e-on-home-solar-and-heating/id1468486784#?platform=ipad"
      >
        <h3>
          E.ON Home for iPad
        </h3>
        <p>
          An award winning smart energy management application for iPad
        </p>
        <code>
          Technologies used:
          <br />
          TypeScript, React, React Native, Redux, MS Azure, AWS, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        responsive={
          Array [
            Object {
              "breakpoint": 768,
              "settings": Object {
                "slidesToShow": 1,
              },
            },
          ]
        }
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="E.ON Home for iPad screenshot 1"
            src="images/temp/eon-home-ipad-dark.jpg"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="E.ON Home for iPad screenshot 2"
            src="images/temp/eon-home-ipad-light.jpg"
          />
        </div>
      </Slider>
    </div>
    <div
      className="c-section__slider"
      key="5"
    >
      <Component
        href="https://apps.apple.com/us/app/foton-home/id1525688620#?platform=ipad"
      >
        <h3>
          Foton Home for iPad
        </h3>
        <p>
          Smart energy management application for iPad
        </p>
        <code>
          Technologies used:
          <br />
          TypeScript, React, React Native, Redux, MS Azure, AWS, NodeJS
        </code>
      </Component>
      <Slider
        autoplaySpeed={5000}
        dots={false}
        infinite={true}
        responsive={
          Array [
            Object {
              "breakpoint": 768,
              "settings": Object {
                "slidesToShow": 1,
              },
            },
          ]
        }
        slidesToScroll={1}
        slidesToShow={2}
        speed={500}
      >
        <div
          key="0"
        >
          <img
            alt="Foton Home for iPad screenshot 1"
            src="images/temp/foton-home-ipad-light.png"
          />
        </div>
        <div
          key="1"
        >
          <img
            alt="Foton Home for iPad screenshot 2"
            src="images/temp/foton-home-ipad-dark.png"
          />
        </div>
      </Slider>
    </div>
  </div>
  <h2>
    Web applications
  </h2>
  <div
    className="c-section__body"
  >
    <Component
      className=""
      href="https://app.home.eon.com/"
      key="0"
    >
      <div>
        <strong>
          E.ON Home
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          TypeScript, SCSS, React, Redux, MS Azure, NodeJS
        </span>
      </div>
      <object
        data="images/temp/eon-home.png"
        title="E.ON Home"
        type="image/png"
        width="600"
      >
        <img
          alt="E.ON Home"
          loading="lazy"
          src="images/temp/E.ON-Home.png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="https://hems-admin-portal.eon.ooo/"
      key="1"
    >
      <div>
        <strong>
          E.ON H.E.M.S (Home Energy Management System)
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          TypeScript, SCSS, Angular, Redux, MS Azure, NodeJS
        </span>
      </div>
      <object
        data="images/temp/eon-hems.png"
        title="E.ON H.E.M.S (Home Energy Management System)"
        type="image/png"
        width="600"
      >
        <img
          alt="E.ON H.E.M.S (Home Energy Management System)"
          loading="lazy"
          src="images/temp/E.ON-H.E.M.S-(Home-Energy-Management-System).png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="https://xpndnow.com/"
      key="2"
    >
      <div>
        <strong>
          XPND
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          TypeScript, SCSS, React, Redux, NodeJS
        </span>
      </div>
      <object
        data="images/temp/xpnd.png"
        title="XPND"
        type="image/png"
        width="600"
      >
        <img
          alt="XPND"
          loading="lazy"
          src="images/temp/xpnd.png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="https://xpndnow.com/ebook"
      key="3"
    >
      <div>
        <strong>
          XPND Ebook
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          TypeScript, SCSS, React, Redux, NodeJS
        </span>
      </div>
      <object
        data="images/temp/xpnd-ebook.png"
        title="XPND Ebook"
        type="image/png"
        width="600"
      >
        <img
          alt="XPND Ebook"
          loading="lazy"
          src="images/temp/xpnd-ebook.png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="https://demo.xpndnow.com/"
      key="4"
    >
      <div>
        <strong>
          XPND demo application
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          TypeScript, SCSS, React, Redux, NodeJS
        </span>
      </div>
      <object
        data="images/temp/xpnd-demo.png"
        title="XPND demo application"
        type="image/png"
        width="600"
      >
        <img
          alt="XPND demo application"
          loading="lazy"
          src="images/temp/XPND-demo-application.png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="http://emailio.com/"
      key="5"
    >
      <div>
        <strong>
          Emailio Web Application
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          SCSS, JavaScript, React, Redux, Redux Saga
        </span>
      </div>
      <object
        data="images/temp/emailio.png"
        title="Emailio Web Application"
        type="image/png"
        width="600"
      >
        <img
          alt="Emailio Web Application"
          loading="lazy"
          src="images/temp/Emailio-Web-Application.png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="https://www.nulla.tv/"
      key="6"
    >
      <div>
        <strong>
          Nulla TV
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          HTML5, CSS3, JavaScript, WordPress
        </span>
      </div>
      <object
        data="images/temp/nulla-tv.png"
        title="Nulla TV"
        type="image/png"
        width="600"
      >
        <img
          alt="Nulla TV"
          loading="lazy"
          src="images/temp/nulla-tv.png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="https://kinetikautomotive.com/"
      key="7"
    >
      <div>
        <strong>
          Kinetik Automotive
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          HTML5, CSS3, JavaScript, Vue, Nuxt, WordPress
        </span>
      </div>
      <object
        data="images/temp/kinetik-automotive.png"
        title="Kinetik Automotive"
        type="image/png"
        width="600"
      >
        <img
          alt="Kinetik Automotive"
          loading="lazy"
          src="images/temp/kinetik-automotive.png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className=""
      href="https://www.sod.bg/"
      key="8"
    >
      <div>
        <strong>
          SOD 64 (СОД 64)
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          HTML5, CSS3, JavaScript, jQuery, PHP
        </span>
      </div>
      <object
        data="images/temp/sod-bg.png"
        title="SOD 64 (СОД 64)"
        type="image/png"
        width="600"
      >
        <img
          alt="SOD 64 (СОД 64)"
          loading="lazy"
          src="images/temp/SOD-64-(СОД-64).png"
          width="600"
        />
      </object>
    </Component>
    <Component
      className="disabled"
      href=""
      key="9"
    >
      <div>
        <strong>
          At the Wall
        </strong>
        <span>
          <small>
            Technologies used:
          </small>
          <br />
          CSS3, PHP
        </span>
      </div>
      <object
        data="images/temp/at-the-wall.png"
        title="At the Wall"
        type="image/png"
        width="600"
      >
        <img
          alt="At the Wall"
          loading="lazy"
          src="images/temp/at-the-wall.png"
          width="600"
        />
      </object>
    </Component>
  </div>
</Component>
`;
